src/web/js/dashboard/GoogleAPI.js

```diff
@@ -93,31 +110,449 @@ class GoogleAPI {
+        name: className,
+        students: [],
+        assignments: []
+      };
+
+      data.nextClassID += 1;
+      // old version was using arrays. updated to use object for easy access via ID
+      // data.classList.push(classInfo)
+      data.classList[classInfo.id] = classInfo;
+
+      //send data back to google
+      return this.savePyretData(data);
+    });
+  }
+
+  /**
+  Attempts to get the class with id = classID. Returns undefined if the id does not exist.
+  */
+  getClass = (classID) => {
+    return this.getPyretData().then((response) => {
+      return response.result.classList[classID];
+    });
+  }
+
+  /**
+  Gets all of this teacher's classes
+  */
+  getAllClasses = () => {
+    return this.getPyretData().then((response) => {
+      return response.result.classList;
+    });
+  }
+
+  /**
+  Removes a class with the given id from the list of classes, if it exists
+  */
+  removeClass = (classID) => {
+    return this.getPyretData().then((response) => {
+      var data = response.result;
+      delete data.classList[classID];
+      return this.savePyretData(data);
+    });
+  }
+
+  /**
+  Sets the object representing the class with id classID to be the object classInfo
+  */
+  updateClass = (classID, classInfo) => {
+    //TODO: validate the class info
+    return this.getPyretData().then((response) => {
+      var data = response.result;
+      if (classID in data.classList){
+        data.classList[classID] = classInfo;
+      }
+      else {
+        //TODO: not sure what's the best way to throw an error here.
+        return undefined;
+      }
+      return this.savePyretData(data);
+    });
+  }
+
+/**
+  Create a new student, with no classes. Students are specified as follows:
+
+  studentInfo:
+  {
+    // the function handles assigning an id
+    firstName: string
+    lastName: string
+    email: string
+  }
+
+  All of these fields are required to execute this method. If one is missing, an error will be thrown
+  */
+  addStudent = (studentInfo) => {
+    // validation
+    for (var key in studentInfo) {
+      if (key !== "firstName" &&
+          key !== "lastName" &&
+          key !== "email"){
+        //TODO: not sure of best way to throw an error
+        return undefined;
+      }
+    }
+
+    return this.getPyretData().then((response) => {
+      //modify data
+      var data = response.result;
+
+      studentInfo.id = data.nextStudentID;
+      studentInfo.classes = [];
+
+      data.nextStudentID += 1;
+
+      //data.studentList.push(studentInfo)
+      data.studentList[studentInfo.id] = studentInfo;
+      //send data back to google
+      return this.savePyretData(data).then(() => {
+        return studentInfo;
+      });
+    });
+  }
+
+  removeStudent = (studentID) => {
+    //needs to remove student from all classes they are in
+    return this.getPyretData().then((response) => {
+      //I'm pretty sure this doesn't work, but it was apparently working for Kevin...
```

Comment:
I think the issue here was something local to me on account of testing with an old version of the pyretinfo.json file so like Kevin says it should be all good now.